scripts/zones/RoMaeve/Zone.lua

```diff
@@ -34,22 +34,32 @@ end
 function onRegionEnter(player, region)
 end
 
+-- Make Ro'Maeve come to life between 6pm and 6am during a full moon
+local moongate1 = GetNPCByID(ID.npc.MOONGATE_OFFSET)
+local moongate2 = GetNPCByID(ID.npc.MOONGATE_OFFSET + 1)
+local vanadielHour = VanadielHour()
```

Comment:
you'll want to define your variables inside onGameHour rather than above/before it, 

I would also keep the original `local hour = VanadielHour()` that was here before, since there are no earth hours here to accidentally confuse it with.

Reply:
Are you sure about this? Wouldn't that just mean more unnecessary requests being made every 144s? Also I feel that kinda goes against what Zach has been pushing were the code itself should be as explicit as possible at all times.

Reply:
The only ones that wouldn't be being called anyway are the 2 getByID, which even though it is probably ok the vars themselves can be inside a conditional to prevent running them when not needed, but when you have them outside the function like this they exist in a larger scope which I did not catch when you showed me in discord (sorry)

proposed adjustment:
```lua
function onGameHour(zone)
    -- Make Ro'Maeve come to life between 6pm and 6am during a full moon
    local romaeveActive = moongate1:getLocalVar("romaeveActive")
    local vanadielHour = VanadielHour()

    if romaeveActive == 0 and IsMoonFull() and (vanadielHour >= 18 or vanadielHour < 6) then
        local moongate1 = GetNPCByID(ID.npc.MOONGATE_OFFSET)
        local moongate2 = GetNPCByID(ID.npc.MOONGATE_OFFSET + 1)

        -- Loop over the affected NPCs: Moongates, bridges and fountain
        for i = ID.npc.MOONGATE_OFFSET, ID.npc.MOONGATE_OFFSET + 7 do
            GetNPCByID(i):setAnimation(tpz.anim.OPEN_DOOR) -- Open them
        end
        moongate2:untargetable(true)
        moongate1:untargetable(true)
        moongate1:setLocalVar("romaeveActive", 1) -- Make this loop unavailable after firing

    -- Clean up at 6am
    elseif romaeveActive == 1 and vanadielHour == 6 then
        local moongate1 = GetNPCByID(ID.npc.MOONGATE_OFFSET)
        local moongate2 = GetNPCByID(ID.npc.MOONGATE_OFFSET + 1)
        for i = ID.npc.MOONGATE_OFFSET, ID.npc.MOONGATE_OFFSET + 7 do
            GetNPCByID(i):setAnimation(tpz.anim.CLOSE_DOOR)
        end
        moongate2:untargetable(false)
        moongate1:untargetable(false)
        moongate1:setLocalVar("romaeveActive", 0) -- Make loop available again
    end
end
```
it won't matter that we're doing a local define of those npc's in both conditions, because only one of the 2 conditions can be trues and each one is inside the scope of that condition (so no collision, and only one set is used). 

hour and romaeveActive are going to be check every hours anyway.

Reply:
I for one like that solution very much. Thank you for your patience @TeoTwawki 
Changes were made as proposed.